src/main/java/io/stargate/sgv2/jsonapi/service/resolver/update/TableUpdateAnalyzer.java

```diff
@@ -0,0 +1,85 @@
+package io.stargate.sgv2.jsonapi.service.resolver.update;
+
+import static io.stargate.sgv2.jsonapi.exception.ErrorFormatters.*;
+import static io.stargate.sgv2.jsonapi.exception.ErrorFormatters.errFmtColumnMetadata;
+
+import com.datastax.oss.driver.api.core.CqlIdentifier;
+import com.datastax.oss.driver.api.core.metadata.schema.ColumnMetadata;
+import com.datastax.oss.driver.api.core.metadata.schema.TableMetadata;
+import io.stargate.sgv2.jsonapi.exception.UpdateException;
+import io.stargate.sgv2.jsonapi.service.cqldriver.executor.TableBasedSchemaObject;
+import io.stargate.sgv2.jsonapi.service.operation.query.ColumnAssignment;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Analyzer for table update command.<br>
+ * For Table UpdateOne update clause there are several rules to check.
+ *
+ * <p>1. checkUnknownColumns(). Can not update on unknown columns. <br>
+ * 2. checkUpdateOnPrimaryKey(). Can not update on PK columns. <br>
+ */
+public class TableUpdateAnalyzer {
+
+  private final TableBasedSchemaObject tableSchemaObject;
+  private final TableMetadata tableMetadata;
+  private final Map<CqlIdentifier, ColumnMetadata> tablePKColumns;
+  private final Map<CqlIdentifier, ColumnMetadata> allColumns;
+
+  public TableUpdateAnalyzer(TableBasedSchemaObject tableSchemaObject) {
+    this.tableSchemaObject =
+        Objects.requireNonNull(tableSchemaObject, "tableSchemaObject cannot be null");
+    tableMetadata = tableSchemaObject.tableMetadata();
+    tablePKColumns =
+        tableMetadata.getPrimaryKey().stream()
+            .collect(Collectors.toMap(ColumnMetadata::getName, Function.identity()));
+    allColumns = tableMetadata.getColumns();
+  }
+
+  public void analyze(List<ColumnAssignment> columnAssignments) {
+    // check rules
+    checkUnknownColumns(columnAssignments);
+    checkUpdateOnPrimaryKey(columnAssignments);
+  }
+
+  /** Update on unknown column is not allowed * */
+  private void checkUnknownColumns(List<ColumnAssignment> columnAssignments) {
+    List<CqlIdentifier> unknownColumns =
+        columnAssignments.stream()
+            .filter(columnAssignment -> !allColumns.containsKey(columnAssignment.column))
+            .map(columnAssignment -> columnAssignment.column)
+            .toList();
```

Comment:
sort the list using our comparator so error messages are stable

Reply:
fixed